AsyncDisplayKit/Details/ASEnvironment.h

```diff
@@ -154,13 +156,15 @@ ASDISPLAYNODE_EXTERN_C_END
   [super setEnvironmentState:environmentState];\
   ASEnvironmentTraitCollection currentTraits = environmentState.traitCollection;\
   if (ASEnvironmentTraitCollectionIsEqualToASEnvironmentTraitCollection(currentTraits, oldTraits) == NO) {\
-    NSArray<NSArray <ASCellNode *> *> *completedNodes = [self.view.dataController completedNodes];\
-    for (NSArray *sectionArray in completedNodes) {\
-      for (ASCellNode *cellNode in sectionArray) {\
-        ASEnvironmentStatePropagateDown(cellNode, currentTraits);\
-        [cellNode setNeedsLayout];\
+    dispatch_async(dispatch_get_main_queue(), ^{\
```

Comment:
adding a dispatch_async is a material change; could you elaborate on why this is required?  It might be fine, or it might introduce cases where additional traffic (e.g. layout passes) are caused.  It would definitely prevent the possibility from piggybacking on any current setNeedsLayout that the cellNode already has scheduled, since any currently-dirtied layout will execute before the block does.